File: java/leetcode/convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return initializeNode(nums, 0, nums.length - 1);
    }
    
    public TreeNode initializeNode(int[] nums, int left, int right) {
        if (left > right) { return null; }
        if (left == right) { return new TreeNode(nums[left]); }
        int midpoint = left + Math.round((right - left) / 2);
        TreeNode root = new TreeNode(nums[midpoint]);
        root.left = initializeNode(nums, left, midpoint - 1);
        root.right = initializeNode(nums, midpoint + 1, right);
        return root;
    }
}

File: javascript/leetcode/convert-sorted-array-to-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    return initializeNode(nums, 0, nums.length - 1);
};

var initializeNode = function(nums, left, right) {
    if (left > right) { return null; }
    if (left === right) { return new TreeNode(nums[left]); }
    var midpoint = left + Math.round((right - left) / 2);
    var root = new TreeNode(nums[midpoint]);
    root.left = initializeNode(nums, left, midpoint - 1);
    root.right = initializeNode(nums, midpoint + 1, right);
    return root;
}
